diff --git a/program/src/ids.rs b/program/src/ids.rs
index 7b4caae..e69de29 100644
--- a/program/src/ids.rs
+++ b/program/src/ids.rs
@@ -1,28 +0,0 @@
-pub mod srm_token {
-    use solana_program::declare_id;
-    #[cfg(feature = "devnet")]
-    declare_id!("AvtB6w9xboLwA145E221vhof5TddhqsChYcx7Fy3xVMH");
-    #[cfg(not(feature = "devnet"))]
-    declare_id!("SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt");
-}
-
-pub mod msrm_token {
-    use solana_program::declare_id;
-    #[cfg(feature = "devnet")]
-    declare_id!("8DJBo4bF4mHNxobjdax3BL9RMh5o71Jf8UiKsf5C5eVH");
-    #[cfg(not(feature = "devnet"))]
-    declare_id!("MSRMcoVyrFxnSgo5uXwone5SKcGhT1KEJMFEkMEWf9L");
-}
-
-pub mod mngo_token {
-    use solana_program::declare_id;
-    #[cfg(feature = "devnet")]
-    declare_id!("Bb9bsTQa1bGEtQ5KagGkvSHyuLqDWumFUcRqFusFNJWC");
-    #[cfg(not(feature = "devnet"))]
-    declare_id!("MangoCzJ36AjZyKwVj3VnYU4GTonjfVEnJmvvWaxLac");
-}
-
-pub mod luna_pyth_oracle {
-    use solana_program::declare_id;
-    declare_id!("5bmWuR1dgP4avtGYMNKLuxumZTVKGgoN2BCMXWDNL9nY");
-}
diff --git a/program/src/matching.rs b/program/src/matching.rs
index 16088cf..35b4862 100644
--- a/program/src/matching.rs
+++ b/program/src/matching.rs
@@ -19,7 +19,6 @@ use mango_logs::{mango_emit_stack, ReferralFeeAccrualLog};
 use mango_macro::{Loadable, Pod};
 
 use crate::error::{check_assert, MangoError, MangoErrorCode, MangoResult, SourceFileId};
-use crate::ids::mngo_token;
 use crate::queue::{EventQueue, FillEvent, OutEvent};
 use crate::state::{
     DataType, MangoAccount, MangoCache, MangoGroup, MetaData, PerpMarket, PerpMarketCache,
@@ -2063,10 +2062,7 @@ fn determine_ref_vars<'a>(
     referrer_mango_account_ai: Option<&'a AccountInfo>,
     now_ts: u64,
 ) -> MangoResult<(I80F48, Option<RefMut<'a, MangoAccount>>)> {
-    let mngo_index = match mango_group.find_token_index(&mngo_token::id()) {
-        None => return Ok((ZERO_I80F48, None)),
-        Some(i) => i,
-    };
+    let mngo_index = 0;
 
     let mngo_cache = &mango_cache.root_bank_cache[mngo_index];
     let tier_2_enabled = mango_group.ref_surcharge_centibps_tier_2 != 0
diff --git a/program/src/processor.rs b/program/src/processor.rs
index 3181140..7bc9789 100644
--- a/program/src/processor.rs
+++ b/program/src/processor.rs
@@ -35,7 +35,6 @@ use mango_logs::{
 };
 
 use crate::error::{check_assert, MangoError, MangoErrorCode, MangoResult, SourceFileId};
-use crate::ids::{luna_pyth_oracle, msrm_token, srm_token};
 use crate::instruction::MangoInstruction;
 use crate::matching::{Book, BookSide, ExpiryType, OrderType, Side};
 use crate::oracle::{determine_oracle_type, OracleType, StubOracle, STUB_MAGIC};
@@ -129,7 +128,6 @@ impl Processor {
             check!(msrm_vault.delegate.is_none(), MangoErrorCode::InvalidVault)?;
             check!(msrm_vault.close_authority.is_none(), MangoErrorCode::InvalidVault)?;
             check_eq!(msrm_vault.owner, mango_group.signer_key, MangoErrorCode::InvalidVault)?;
-            check_eq!(&msrm_vault.mint, &msrm_token::ID, MangoErrorCode::InvalidVault)?;
             check_eq!(msrm_vault_ai.owner, &spl_token::ID, MangoErrorCode::InvalidVault)?;
             mango_group.msrm_vault = *msrm_vault_ai.key;
         }
@@ -508,14 +506,6 @@ impl Processor {
             mango_group.tokens[QUOTE_INDEX].mint.to_aligned_bytes(),
             MangoErrorCode::Default
         )?;
-
-        // TODO - what if quote currency is mngo, srm or msrm
-        // if mint is SRM set srm_vault
-
-        if mint_ai.key == &srm_token::ID {
-            check!(mango_group.srm_vault == Pubkey::default(), MangoErrorCode::Default)?;
-            mango_group.srm_vault = *vault_ai.key;
-        }
         Ok(())
     }
 
@@ -8155,11 +8145,6 @@ pub fn read_oracle(
                     conf.to_num::<f64>()
                 );
 
-                // For luna, to prevent market from getting stuck, just continue using last known price in cache
-                if oracle_ai.key == &luna_pyth_oracle::ID {
-                    return Ok(last_known_price_in_cache);
-                }
-
                 return Err(throw_err!(MangoErrorCode::InvalidOraclePrice));
             }
 
diff --git a/program/src/state.rs b/program/src/state.rs
index 2b79ab7..c06b046 100644
--- a/program/src/state.rs
+++ b/program/src/state.rs
@@ -24,7 +24,6 @@ use mango_common::Loadable;
 use mango_macro::{Loadable, Pod, TriviallyTransmutable};
 
 use crate::error::{check_assert, MangoError, MangoErrorCode, MangoResult, SourceFileId};
-use crate::ids::mngo_token;
 use crate::matching::{Book, LeafNode, OrderType, Side};
 use crate::queue::{EventQueue, EventType, FillEvent};
 use crate::utils::{
@@ -1197,7 +1196,7 @@ impl HealthCache {
             let taker_quote_native =
                 I80F48::from_num(info.quote_lot_size.checked_mul(taker_quote.abs()).unwrap());
             let mut market_fees = info.taker_fee * taker_quote_native;
-            if let Some(mngo_index) = mango_group.find_token_index(&mngo_token::id()) {
+            if let Some(mngo_index) = Some(0) {
                 let mngo_cache = &mango_cache.root_bank_cache[mngo_index];
                 let mngo_deposits = mango_account.get_native_deposit(mngo_cache, mngo_index)?;
                 let ref_mngo_req = I80F48::from_num(mango_group.ref_mngo_required);
@@ -2235,7 +2234,6 @@ impl PerpMarket {
         check!(vault.owner == mango_group.signer_key, MangoErrorCode::InvalidOwner)?;
         check!(vault.delegate.is_none(), MangoErrorCode::InvalidVault)?;
         check!(vault.close_authority.is_none(), MangoErrorCode::InvalidVault)?;
-        check!(vault.mint == mngo_token::ID, MangoErrorCode::InvalidVault)?;
         check!(mngo_vault_ai.owner == &spl_token::ID, MangoErrorCode::InvalidOwner)?;
         state.mngo_vault = *mngo_vault_ai.key;
 
